### PR TITLE
[Multi-agent] Define component ownership and dependency graph

### LINKED ISSUE #62: [Multi-agent] Component inventory, ownership, and dependency graph
Parent epic: #60. Inventory and classify shared versus runtime-specific files, including AGENTS.md, .agents/skills, OpenAI metadata, CLAUDE.md, hooks, setup content, and docs. Define composable components and dependencies with exactly one owner per generated output path. Acceptance: checked ownership manifest covers every managed path; shared-file collisions are rejected; clean-composition prerequisites are documented. Can proceed alongside the registry work.

### PR DESCRIPTION
**Pull request by conorbronsdon**
## What's in this PR

Implements #62 as the second foundation slice of #60.

- adds an authoritative eight-component inventory with exact managed, seed, and development ownership for 167 tracked paths
- validates dependencies, deterministic closure, bidirectional Git coverage, symlinks, traversal, Unicode/case collisions, Windows aliases, reserved local state, and generated-schema drift
- separates portable skill bodies from OpenAI discovery metadata
- cross-validates runtime instructions, skills, onboarding docs, descriptors, and conformance evidence against selected component closures
- documents the read-only inventory boundary and the remaining gates before clean materialization

This PR is intentionally stacked on #75 (base: feat/runtime-registry-v2). It does not copy, delete, upgrade, or slim workspace files.

## Review and verification

Two review rounds completed with Claude Opus, Hermes free-model routing (stealth/ox-alpha first; openrouter/free fallback after a 429), and two independent GPT-5.6-sol lenses. All final reviewers reported no blockers.

- 215 Python tests passed; 12 skips (11 opt-in runtime launches plus one Windows symlink-privilege skip)
- 32 portability tests passed
- 12 hook tests passed
- integration catalog, component inventory, runtime registry, links, generated artifacts, and whitespace checks passed
- component inventory: 8 components, 167 exact paths
- Bash aggregate runner reaches the known Windows/WSL linked-worktree Git-path limitation; native Windows Python and the independent Bash portability/hook suites pass

### Skills & commands
- [x] No new skills or commands are introduced
- [x] Portable workflows remain provider-neutral

### Repo hygiene
- [x] No sensitive data committed
- [x] CHANGELOG.md updated
- [ ] CI passes

Closes #62

### CODE REVIEW

File: contextos/component_schema.py [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+        _identity(path_entry["path"])
+        for component in document["components"]
+        for path_entry in component["paths"]
+    }
+    extensible = [tuple(_identity(path).split("/"))
+                  for path in document["extensible_roots"]]
+    seen: dict[str, str] = {}
+    missing: list[str] = []
+    for index, value in enumerate(tracked_paths):
+        path = _ownership_path(value, f"tracked_paths[{index}]")
```
Comment (conorbronsdon) [inline contextos/component_schema.py]: **Minor — workspace-owned personal files get manifest-grade path policing, and failures don't name the file.** The docs say new user files are workspace-owned rather than component-owned, and README directs users to run `validate-all.sh --workspace` after adding personal context — but `unclassified_tracked_paths` runs every git-tracked path through `_ownership_path(value, f"tracked_paths[{index}]")` *before* the extensible-root exemption is consulted. Confirmed by execution: a legal Linux/macOS personal file `inbox/what next?.md` under the extensible root `inbox` hard-fails the entire check with `tracked_paths[0]: must not use a Windows-aliased or illegal path segment`, and `sessions/2026-08-25.log` fails with `must not claim local, generated-on-demand, secret, or ignored state`. Worse, `_fail` reports only the index into unsorted `git ls-files` output, so the user has to count entries to find the offender. Workspace-owned files the inventory explicitly does not own should be exempt from the Windows-alias/extension policing (or at minimum the error should include the path).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/component-manifests.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+
+def schema_text() -> str:
+    return json.dumps(component_schema_document(), indent=2, ensure_ascii=False) + "\n"
+
+
+def git_tracked_paths(root: Path) -> list[str]:
+    try:
+        result = subprocess.run(
+            ["git", "ls-files", "-z", "--"],
```
Comment (conorbronsdon) [inline scripts/component-manifests.py:37]: **Minor — conflicted checkouts produce a misleading "collision" error.** `git ls-files -z --` without `--deduplicate` lists a conflicted path once per merge stage, and `unclassified_tracked_paths` then aborts with the self-referential `portable path collision: 'inbox/a.md' and 'inbox/a.md'` (confirmed by direct call with a duplicated entry) — reported as a Unicode/case collision when the real problem is a merge in progress. `--deduplicate` (git ≥ 2.31) or a Python-side dedupe before the collision check keeps that error meaningful.

Related test gap: the script layer has no tests at all — `git_tracked_paths` (unterminated-output and non-UTF-8 branches), `check()`'s schema-staleness comparison, and the `--allow-extensible`-only-with-`check` CLI rule are exercised only implicitly by `validate-all.sh` on a clean repo; all 18 unit tests drive the library with synthetic path lists.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/test_runtime_manifests.py [hunk 3]
```diff
@@ -173,6 +216,41 @@ def test_new_descriptor_is_discovered_without_code_change(self) -> None:
...
+            core_paths.remove({"path": "docs/onboarding.md", "policy": "managed"})
+            unselected_owner["components"].append({
+                "id": "docs-extra", "description": "Unselected docs fixture.",
+                "depends_on": ["core"],
+                "paths": [{"path": "docs/onboarding.md", "policy": "managed"}],
+            })
+            (root / "components/manifest.json").write_text(
+                json.dumps(unselected_owner), encoding="utf-8"
+            )
+            with self.assertRaisesRegex(ContextOSError, "unselected components"):
```
Comment (conorbronsdon) [inline tests/test_runtime_manifests.py:252]: **Minor — the directory-descendant ownership branch has no negative test.** This unselected-components test covers only an exact-file required path (`docs/onboarding.md`), and the fixture's instruction/skill sources are all `workspace`-scoped, so the descendant-matching branch in `runtime_manifest` (`portable_path_identity(owned_path).startswith(required_identity + "/")`) is never covered by a test that asserts failure — only incidentally by the real descriptors validating. A regression there (e.g. dropping the `"/"` boundary so `docs-x` matches a required `docs`) would slip through.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: contextos/kernel.py [hunk 4]
```diff
@@ -613,7 +626,44 @@ def runtime_manifest(
...
+            if not matching_owners:
+                raise ComponentManifestError(
+                    f"runtime {runtime!r} references unowned path {required_path!r}"
+                )
+            if not matching_owners.intersection(selected):
+                raise ComponentManifestError(
+                    f"runtime {runtime!r} requires {required_path!r}, owned only by "
+                    f"unselected components {sorted(matching_owners)!r}"
+                )
+    except (RuntimeManifestError, ComponentManifestError, OSError, UnicodeError) as exc:
```
Comment (conorbronsdon) [inline contextos/kernel.py:666]: **Nit — inventory failures are blamed on the wrong artifact.** All `ComponentManifestError`s get wrapped as `invalid runtime manifest: {manifest_path} (...)`. Confirmed by execution: deleting a path from `components/manifest.json` yields `invalid runtime manifest: runtimes/codex.json (runtime 'codex' references unowned path ...)`, and a malformed `components/manifest.json` likewise blames `runtimes/<r>.json`. The broadened except also makes a missing/corrupt component inventory fatal to every `runtime_manifest` call, hooks included — intended per the docstring, but a distinct `invalid component inventory:` prefix would point maintainers at the real file.

Also a tiny dead-surface note: `dependency_closure = component_closure` (component_schema.py:398) has zero callers on the head tree, and the `location` element passed to `_reject_normalized_duplicates` is stored but never used in an error message.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by conorbronsdon — COMMENTED** [review]
Reviewed the incremental diff (`feat/runtime-registry-v2..feat/component-inventory`) with full head-branch context, and verified the central claim directly: `git ls-tree -r` at head yields exactly 167 files and the manifest owns exactly those 167 paths — both difference sets empty. `scripts/component-manifests.py check`, the new unit tests, and full `validate-all.sh` all pass at head in a scratch clone.

**Overall:** carefully engineered. The closure/cycle logic, adjacent-pair prefix-conflict detection (provably sufficient after segment-wise sorting), NFC+casefold identity, symlink-ancestor walks, and the generate/check drift gate all held up under attempts to break them. The docs' "read-only inventory" and "directory sources may contain unselected fragments" claims accurately describe the code. Remaining issues are operational edge cases in the new `--workspace` mode plus diagnostics/test gaps — no blockers. Four inline comments.

Verified non-issues, for the record: partial-selection semantics for directory sources are explicitly documented; conformance tests being `development`-policy while runtimes require them is covered by clean-composition prerequisites 7–8; the claude closure omitting `agents-instructions` despite the AGENTS.md doctor requirement is acknowledged in prerequisite 9; and the epic's one-owner-per-path contract holds — no double-ownership, and the OpenAI metadata split (`.agents/skills/*/agents/openai.yaml` → `openai-skill-metadata`, reached by codex via `codex-adapter`'s `depends_on`) matches the required graph.

---
_Generated by [Claude Code](https://claude.ai/code)_
